db/schema.rb

```diff
@@ -175,4 +175,50 @@
        JOIN assignments ass ON ((ass.id = app.id)))
        JOIN defendants def ON ((def.id = app.id)));
   SQL
+  create_view "active_providers", sql_definition: <<-SQL
+      WITH submissions AS (
+           SELECT application.application_type,
+              ((application_version.application -> 'firm_office'::text) ->> 'account_number'::text) AS office_code,
```

Comment:
so office code is not always populated and may not be correct from what I can see

```
    def initialize_application(attributes = {}, &block)
      attributes.merge!(
        office_code: (current_provider.office_codes.first unless current_provider.multiple_offices?),
        submitter: current_provider,
        laa_reference: generate_laa_reference
      )
      Claim.create!(attributes).tap(&block)
    end
```

Reply:
Ah, so at the point the claim is submitted, `office_code` is guaranteed to be populated because if there are multiple office codes the manual office code selection form has to be submitted. 
(If it isn't populated, neither will firm_office->account_number be, because at the point of payload generation they both come from the same database field.)

There may be old dodgy dev data where the field is missing, but for all intents and purposes you should be able to count on it.

Reply:
makes sense, let me have a look and confirm on the state of the data (may add rake task to fix if required. :)